FinancialTracker/Firebase/AuthManager.swift

```diff
@@ -101,126 +99,148 @@ class AuthManager {
-                    case .access(let error):
-                        completionHandler(FirebaseError.access(error), user)
-                    case .auth, .signOut, .nonExistingUser:
-                        assertionFailure("This error should not appear: \(firebaseError.localizedDescription)")
-                        // swiftlint:disable:next unneeded_break_in_switch
-                        break
-                    }
-                } else {
-                    completionHandler(nil, user)
+            self.databaseManager.getCurrentUserData(uid: result.user.uid) { databaseError, user in
+                if let databaseError = databaseError {
+                    completionHandler(AuthError.database(databaseError), nil)
+                    return
                 }
+                completionHandler(nil, user)
             }
         }
     }
     
     /// Signing out the current user
     /// - Parameters:
     ///   - completionHandler: Block that is to be executed if an error appears or the function is successfully executed.
-    ///   - firebaseError: An error object that indicates why the function failed, or nil if the was successful.
-    ///   - user: The user that was returned by firestore.
-    func signOut(completionHandler: @escaping (FirebaseError?, Bool) -> Void) {
+    ///     1. `authError` - An error object that indicates why the function     failed, or nil if the was successful.
+    ///     2. `user` - The user that was returned by firestore.
+    func signOut(completionHandler: @escaping (AuthError?, Bool) -> Void) {
         do {
             try auth.signOut()
-            self.databaseManager.removeFCMTokenFromCurrentUser { firebaseError, success in
+            self.databaseManager.removeFCMTokenFromCurrentUser { databaseError, success in
+                if let databaseError = databaseError {
+                    completionHandler(AuthError.database(databaseError), false)
+                    return
+                }
                 self.databaseManager.setUserToNil()
-                completionHandler(firebaseError, success)
+                completionHandler(nil, success)
             }
         } catch let signOutError {
-            completionHandler(FirebaseError.signOut(signOutError), false)
+            completionHandler(AuthError.signOut(signOutError), false)
         }
     }
     
-    func addBalanceToCurrentUser(_ balance: Double, currency: Currency, completionHandler: @escaping (FirebaseError?, Bool) -> Void) {
-        databaseManager.addBalanceToCurrentUser(balance, currency: currency) { firebaseError, success in
-            completionHandler(firebaseError, success)
+    func addBalanceToCurrentUser(_ balance: Double, currency: Currency, completionHandler: @escaping (AuthError?, Bool) -> Void) {
+        databaseManager.addBalanceToCurrentUser(balance, currency: currency) { databaseError, success in
+            if let databaseError = databaseError {
```

Comment:
No need for the `if let` here, just bubble the completion up because your completionHandler passes up optional `AuthError?`: 

`completionHandler(AuthError.database(databaseError), success)`

You will need to add a failable init to `AuthError` enum

Reply:
The same applies to all similar situations.